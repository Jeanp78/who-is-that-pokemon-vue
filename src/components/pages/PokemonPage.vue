<template>
  <div v-if="!pokemonAnswer">
    <h1>ESPERE UN MOMENTO</h1>
    <img src="@/assets/loading.jpeg" alt="" width="500" />
  </div>

  <div v-else class="fade-in">
    <h1>¿Quién es ese pokemon?</h1>

    <!--IMAGEN-->
    <PokemonPictures :pokemonId="pokemonAnswer.id" :showPokemon="showPokemon" />
    <!--OPCIONES-->
    <PokemonOptions :options="pokemonArr" @selection="checkAnswer($event)" />
    <div v-if="showAnswer">
    <h2>{{ message }}</h2>
    <button @click="newGame">Jugar de nuevo</button>
    </div>
  </div>
</template>

<script>
import PokemonPictures from "../PokemonPictures.vue";
import PokemonOptions from "../PokemonOptions.vue";
import getPokemonOptions from "@/helpers/getPokemonOptions";

// getPokemonOptions().then(console.log)

export default {
  data() {
    return {
      pokemonArr: [],
      pokemonAnswer: null,
      showPokemon: false,
      message: "",
      showAnswer : false
    };
  },
  methods: {
    async mixPokemon() {
      this.pokemonArr = await getPokemonOptions();
      const rndInt = Math.floor(Math.random() * 4);
      this.pokemonAnswer = this.pokemonArr[rndInt];

      //  console.log(this.pokemonImg)
      //  console.log(this.pokemonArr)
    },
    checkAnswer(id) {
      this.showPokemon = true;
        this.showAnswer = true
      if (id === this.pokemonAnswer.id) {
        this.message = `Correcto, el pokemon es ${this.pokemonAnswer.name}`;
      } else {
        this.message = `Opss, el pokemon era ${this.pokemonAnswer.name}`;
      }
      console.log(`boton clickeado y el id ${id}`);
    },
    newGame() {
      this.mixPokemon();
      this.showPokemon = false;
      this.message = "";
      this.pokemonArr=[]
      this.pokemonAnswer = null
      this.showAnswer = false 
    },
  },
  mounted() {
    this.mixPokemon();
  },
  components: {
    PokemonPictures,
    PokemonOptions,
  },
};
</script>

