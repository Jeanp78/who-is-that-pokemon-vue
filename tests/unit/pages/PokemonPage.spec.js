import { mount, shallowMount } from "@vue/test-utils";
import PokemonPage from '@/components/pages/PokemonPage'
import { pokemons } from "../mocks/pokemons.mock";

describe('Pruebas sobre Pokemon Page', () => {

    let wrapper
    beforeEach(() => {
        wrapper = shallowMount(PokemonPage)
    })


    test('Snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('Se debe llamar el mixedPokemons', () => {

        const mixPokemonArraySpy = jest.spyOn(PokemonPage.methods, 'mixPokemon')
        const wrapper = shallowMount(PokemonPage)

        expect(mixPokemonArraySpy).toHaveBeenCalled()

    })

    test('Debe hacer el snapshot cuando cargan los componentes', () => {
        const wrapper = shallowMount(PokemonPage, {
            data() {
                return {
                    pokemonArr: pokemons,
                    pokemonAnswer: pokemons[0],
                    showPokemon: false,
                    message: "",
                    showAnswer: false
                }
            }
        })

        expect(wrapper.html()).toMatchSnapshot()

    })

    test('debe de mostrar los componentes de PokemonPicture y PokemonOptions', () => {

        const wrapper = shallowMount(PokemonPage, {
            data() {
                return {
                    pokemonArr: pokemons,
                    pokemonAnswer: pokemons[0],
                    showPokemon: false,
                    message: "",
                    showAnswer: false
                }
            }
        })

        const picture = wrapper.find('pokemon-pictures-stub')
        const options = wrapper.find('pokemon-options-stub')

        expect(picture.exists()).toBeTruthy()
        expect(options.exists()).toBeTruthy()


        expect(picture.attributes('pokemonid')).toBe('5')
        expect(options.attributes('options')).toBeTruthy()


    })

    test('pruebas con checkAnswer', async () => {

        const wrapper = shallowMount(PokemonPage, {
            data() {
                return {
                    pokemonArr: pokemons,
                    pokemonAnswer: pokemons[0],
                    showPokemon: false,
                    message: "",
                    showAnswer: false
                }
            }
        })

        await wrapper.vm.checkAnswer(5)

        expect(wrapper.find('h2').exists()).toBeTruthy()
        expect(wrapper.vm.showPokemon).toBe(true)
        expect(wrapper.find('h2').text()).toBe(`Correcto, el pokemon es ${pokemons[0].name}`)
        

        await wrapper.vm.checkAnswer(10)

        expect(wrapper.vm.message).toBe(`Opss, el pokemon era ${pokemons[0].name}`)


    })


})
