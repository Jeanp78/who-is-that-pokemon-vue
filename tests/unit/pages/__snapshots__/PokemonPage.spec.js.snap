// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre Pokemon Page Debe hacer el snapshot cuando cargan los componentes 1`] = `
<div class="fade-in">
  <h1>¿Quién es ese pokemon?</h1>
  <!--IMAGEN-->
  <pokemon-pictures-stub pokemonid="5" showpokemon="false"></pokemon-pictures-stub>
  <!--OPCIONES-->
  <pokemon-options-stub options="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
  <!--v-if-->
</div>
`;

exports[`Pruebas sobre Pokemon Page Snapshot 1`] = `
<div>
  <h1>ESPERE UN MOMENTO</h1><img src="" alt="" width="500">
</div>
`;
